### ISSUE TITLE
Regexp parsing fails

### ISSUE THREAD

**Issue by raynigon** [tmont/nginx-conf#20]
Hi @tmont would be cool if you could fix this, since a user of the VSCode Formatting Plugin ran into this bug. The connected issue is raynigon/vscode-nginx-formatter#4

When parsing
```
location ~ ^/\d{3}/ {
    try_files $uri /index.html;
}
```
the dumped string looks like
```
location ~ ^/\d;
```

**Comment by raynigon** [comment]
@tmont do you still maintain this package ?

**Comment by tmont** [comment]
I just tested that configuration on `nginx/1.14.0 (Ubuntu)` and it results in an error:

```
nginx: [emerg] unknown directive "3}/"
```

So in this case the parser is doing the right thing and trying to parse the brackets normally. The real nginx parser is aware of all valid directives hence the failure. This library only parses and does not validate, and it will try to assume the configuration is valid and print what it can, hence the truncated location.

Basically your user's configuration is using an incorrect syntax, and s/he probably needs to quote the regular expression.
